### PR TITLE
Animation enhancement and revisions

### PR DESCRIPTION
**Pull request by sg00dwin**
Addition of an animate-slide class, giving affordance transition to specific elements on the overview and refactor of existing component-animations.less

The animate-slide applies brief movement to specific elements on the overview when they enter or exit so that surrounding elements don't abruptly jump in location. It is combined with a fade in/out opacity change that is slight faster, so that these transition are visually smoother.

Elements that have movement are builds, build-pipeline-expanded and service group alerts (pf-toast)

### CODE REVIEW

File: app/styles/_overview.less [hunk 1]
```diff
@@ -128,7 +128,7 @@
   .service-group-header,
   .deployment-tile,
   .standalone-service .builds-block,
-  .service-group-triggers {
+  .service-group-triggers .build {
```
Comment (spadgett) [inline app/styles/_overview.less]: Does this cause multiple shadows when there is more than one build or pipeline?

File: app/views/_alerts.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<div ng-attr-row="{{toast}}"  ng-attr-wrap="{{toast}}">
-  <div ng-repeat="alert in (alerts | filterCollection : filter)" ng-if="!alert.hidden" class="alert-wrapper">
+<div ng-attr-row="{{toast}}"  ng-attr-wrap="{{toast}}" style="overflow: hidden">
```
Comment (spadgett) [inline app/views/_alerts.html]: Let's pull the style into a less file somewhere.

File: app/views/_alerts.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
+  <div ng-repeat="(alertID, alert) in (alerts | filterCollection : filter) track by (alertID + (alert.message || alert.details))" ng-if="!alert.hidden" class="alert-wrapper animate-repeat animate-slide" ng-class="{'animate-slide': has-slide}">
```
Comment (spadgett) [inline app/views/_alerts.html]: You have `animate-slide` in both `class` and `ng-class` here.

File: app/scripts/directives/alerts.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ angular.module('openshiftConsole')
         alerts: '=',
         // Filter function (optional).
         filter: '=?',
+        hasSlide: '=?',
```
Comment (spadgett) [inline app/scripts/directives/alerts.js]: `animateSlide` makes it more clear what this is.

File: app/views/_alerts.html (hunk 2 above)
Comment (sg00dwin) [inline app/views/_alerts.html]: This was removed

File: app/views/_alerts.html (hunk 3 above)
Comment (sg00dwin) [inline app/views/_alerts.html]: Removed the one from within `class`

**Comment by sg00dwin** [comment]
@jwforres @spadgett PTAL

**Comment by spadgett** [comment]
[merge]

**Pull request merged**